src/test/java/jimmy/mcgymmy/logic/predicate/FoodContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,68 @@
+package jimmy.mcgymmy.logic.predicate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import jimmy.mcgymmy.model.food.Name;
+import jimmy.mcgymmy.testutil.FoodBuilder;
+
+class FoodContainsKeywordsPredicateTest {
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        FoodContainsKeywordsPredicate firstPredicate = new FoodContainsKeywordsPredicate(firstPredicateKeywordList);
+        FoodContainsKeywordsPredicate secondPredicate = new FoodContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        FoodContainsKeywordsPredicate firstPredicateCopy = new FoodContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different food -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_nameFoundTagNotFound_returnsTrue() {
+        FoodContainsKeywordsPredicate predicate =
+                new FoodContainsKeywordsPredicate(Arrays.asList("Alice", "Dinner"));
+        assertTrue(predicate.test(new FoodBuilder().withName(new Name("Alice Bob")).withTags("Lunch").build()));
+    }
+
+    @Test
+    public void test_nameFoundTagFound_returnsTrue() {
+        FoodContainsKeywordsPredicate predicate =
+                new FoodContainsKeywordsPredicate(Arrays.asList("Alice", "Dinner"));
+        assertTrue(predicate.test(new FoodBuilder().withName(new Name("Alice Bob")).withTags("Dinner").build()));
+    }
+
+    @Test
+    public void test_nameNotFoundTagFound_returnsTrue() {
+        FoodContainsKeywordsPredicate predicate =
+                new FoodContainsKeywordsPredicate(Arrays.asList("Alice", "Dinner"));
+        assertTrue(predicate.test(new FoodBuilder().withName(new Name("Bob the barber")).withTags("Dinner").build()));
+    }
+
+    @Test
+    public void test_nameNotFoundTagNotFound_returnsFalse() {
+        FoodContainsKeywordsPredicate predicate =
+                new FoodContainsKeywordsPredicate(Arrays.asList("Alice", "Lunch"));
+        assertFalse(predicate.test(new FoodBuilder().withName(new Name("Bob the barber")).withTags("Dinner").build()));
```

Comment:
Might want to change like the names to something more food-related ><

Reply:
The tag search is already case sensitive.
Ok I will change the name

Reply:
Sorry I meant case insensitive ><

Reply:
Isn't making the tag search case sensitive better for categorizing food items? The user can search for exactly one tag that they want.

Reply:
Oo yea true. rip idk what i was thinking when i vetted x.x

Reply:
Haha so i mark this as resolved now